src/ast.cpp

```diff
@@ -391,10 +401,10 @@ namespace Sass {
     return true;
   }
 
-  SimpleSequence_Selector* Simple_Selector::unify_with(SimpleSequence_Selector* rhs, Context& ctx)
+  Compound_Selector_Ptr Simple_Selector::unify_with(Compound_Selector_Ptr rhs, Context& ctx)
   {
     for (size_t i = 0, L = rhs->length(); i < L; ++i)
-    { if (to_string(ctx.c_options) == (*rhs)[i]->to_string(ctx.c_options)) return rhs; }
+    { if (to_string(ctx.c_options) == rhs->at(i)->to_string(ctx.c_options)) return rhs; }
```

Comment:
There is a caveat with `at`: http://stackoverflow.com/questions/3269809/stdvectorat-vs-operator-surprising-results-5-to-10-times-slower-f. Did not do any realworld performance assessements yet. We could also opt to use `[]` in our `at` implementation. But this would make most sense with a complete refactor of List handling in our AST nodes. Already did some tests in that direction.

IMO in the end we should aim to use OO style as RVO should optimize these calls.